pkg/types/compare.go

```diff
@@ -84,3 +85,27 @@ func VecCompareIU(x []int64, y []uint64, res []int64) {
 func CompareString(x, y, collation string) int {
 	return collate.GetCollator(collation).Compare(x, y)
 }
+
+// CompareInt return an integer comparing the integer x to y with signed or unsigned.
+func CompareInt(arg0 int64, isUnsigned0 bool, arg1 int64, isUnsigned1 bool) int {
+	var res int
+	switch {
+	case isUnsigned0 && isUnsigned1:
+		res = cmp.Compare(uint64(arg0), uint64(arg1))
```

Comment:
Why use cmp.Compare here? Just curious since it has a small overhead trying to check for NaN and have some generic code for handling Floats, which is not needed for ints.

Reply:
Just copy it from `pkg/expression/builtin_compare.go`. The same code can be used in three places.

Reply:
OK, I just checked and it has only a minor performance difference, like a few percentage difference.